tf2_ros/src/tf2_echo.cpp

```diff
@@ -103,8 +103,15 @@ int main(int argc, char ** argv)
   std::string source_frameid = std::string(argv[1]);
   std::string target_frameid = std::string(argv[2]);
 
-  // Wait for up to one second for the first transforms to become avaiable. 
-  echoListener.buffer_.canTransform(source_frameid, target_frameid, tf2::TimePoint(), tf2::durationFromSec(1.0));
+  // Wait for up to one second for the first transforms to become avaiable.
+  std::string warning_msg;
+  while (rclcpp::ok() && !echoListener.buffer_.canTransform(
+    source_frameid, target_frameid, tf2::TimePoint(), &warning_msg))
+  {
+    RCLCPP_INFO_THROTTLE(nh->get_logger(), *clock, 1000, "Waiting for transfrom %s ->  %s: %s",
```

Comment:
```suggestion
    RCLCPP_INFO_THROTTLE(nh->get_logger(), *clock, 1000, "Waiting for transform %s ->  %s: %s",
```

Reply:
transformed transfrom to transform in 837170743afc9e51565fcaf0a256b90bdb4a5c80